go-controller/pkg/ovn/namespace_test.go

```diff
@@ -17,20 +17,24 @@ import (
 	. "github.com/onsi/gomega"
 )
 
-func newNamespaceMeta(namespace string) metav1.ObjectMeta {
+func newNamespaceMeta(namespace string, additionalLabels map[string]string) metav1.ObjectMeta {
+	labels := map[string]string{
+		"name": namespace,
+	}
+	for k, v := range additionalLabels {
+		labels[k] = v
+	}
 	return metav1.ObjectMeta{
-		UID:  types.UID(namespace),
-		Name: namespace,
-		Labels: map[string]string{
-			"name": namespace,
-		},
+		UID:         types.UID(namespace),
+		Name:        namespace,
+		Labels:      labels,
 		Annotations: map[string]string{},
 	}
 }
 
-func newNamespace(namespace string) *v1.Namespace {
+func newNamespace(namespace string, additionalLabels map[string]string) *v1.Namespace {
```

Comment:
why do we need labels on the namespace? maybe that will become more clear as i get to the follow up commits

Reply:
As I mentioned here: https://github.com/ovn-org/ovn-kubernetes/pull/1484/#discussion_r458666824 we need to support matching on namespaces, and they way we do that is by using `metav1.LabelSelector` which means the user will need to label namespaces when he/she wants things to match on the namespace/pod he/she wants. And if we want to write tests for this: we'll need this.